sources/directconnect/direct-connect-gateway-attachment_test.go

```diff
@@ -0,0 +1,71 @@
+package directconnect
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func TestDirectConnectGatewayAttachmentOutputMapper(t *testing.T) {
+	output := &directconnect.DescribeDirectConnectGatewayAttachmentsOutput{
+		DirectConnectGatewayAttachments: []types.DirectConnectGatewayAttachment{
+			{
+				VirtualInterfaceId:     sources.PtrString("dxvif-ffhhk74f"),
+				DirectConnectGatewayId: sources.PtrString("cf68415c-f4ae-48f2-87a7-3b52cexample"),
+				AttachmentState:        "detaching",
+			},
```

Comment:
This object also has a [`StateChangeError`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect@v1.22.7/types#DirectConnectGatewayAttachment) attribute. We should add this to the test and set the item health based on it. If there is no error it's healthy, otherwise unhealthy. Here's [an example](https://github.com/overmindtech/aws-source/blob/6a5303519230fb0ae5130bf87dcc68d92e5a0e79/sources/ecs/task.go#L64) of how to do that

Reply:
https://github.com/overmindtech/aws-source/pull/373/commits/adfa94416855a9a6bb34e1aee416af4253777c52